networkit/vizbridges.py

```diff
@@ -4,332 +4,562 @@
+        for i in G.iterNodes():
+            hcColors.append((color[0] * 255, color[1] * 255, color[2] * 255))
+
+    return hcColors
+
+
+def _calculateEdgeColoring(G, palette, edgeScores: List[float] = None):
+
+    # Color calculation: score = continuous distribution, partition = discrete distribution
+    hcColors = []
+
+    # Score
+    if edgeScores:
+
+        minhc = min(edgeScores)
+        maxhc = max(edgeScores)
+
+        # Calculate coloring of nodes
+        def getRgb(minimum, maximum, value):
+            minimum, maximum, value = float(minimum), float(maximum), float(value)
+            ratio = int(
+                (len(palette) - 1)
+                * (value - minimum)
+                / (maximum - minimum)
+                * (value - minimum)
+                / (maximum - minimum)
+            )
+            r = int(palette[ratio][0] * 255)
+            g = int(palette[ratio][1] * 255)
+            b = int(palette[ratio][2] * 255)
+            return r, g, b
+
+        if abs(maxhc - minhc) > 0:
+            for score in edgeScores:
+                hcColors.append(getRgb(minhc, maxhc, score))
+        else:
+            color = palette[int((len(palette) - 1) / 2)]
+            for i in range(0, len(edgeScores)):
+                hcColors.append((color[0] * 255, color[1] * 255, color[2] * 255))
+
+    # No edge values
+    else:
+        color = palette[0]
+        hcColors = [
+            (color[0] * 255, color[1] * 255, color[2] * 255)
+        ] * G.numberOfEdges()
+
+    return hcColors
+
+def _getEdgeScoreDirected(
+    edgeScoresDict: Mapping[Tuple[int, int], float], u: int, v: int
+) -> Optional[float]:
+
+    getUV = edgeScoresDict.get((u,v))
+
+    # missing edge score
+    if getUV == None:
+        raise Exception(f"edgeScores should include scores for every edge. ({u},{v}) is missing");
+
+    return getUV
+
+def _getEdgeScoreUndirected(
+    edgeScoresDict: Mapping[Tuple[int, int], float], u: int, v: int
+) -> Optional[float]:
+
+    getUV = edgeScoresDict.get((u,v))
+    getVU = edgeScoresDict.get((v,u))
+
+    # missing edge score
+    if getUV == None and getVU == None:
+        raise Exception(f"edgeScores should include scores for every edge. ({u},{v}) is missing");
+
+    # (u,v) is present
+    if getUV == None and getVU != None:
+        return getVU
+
+    # (v,u) is present
+    if getVU == None and getUV != None:
+        return getUV
+
+    # both are present and the same
+    if getVU == getUV:
+        return getUV
+    
+    # both directions are present and the score is different (and not None)
+    if isinstance(edgeScoresDict, defaultdict):
+        defaultValue = edgeScoresDict.default_factory()
+        if getUV != defaultValue and getVU != defaultValue:
+            raise Exception(f"edgeScore different for both directions of ({u},{v}).");
+        elif getUV != defaultValue:
+            return getVU
+        else:
+            return getUV
+
+def widgetFromGraph(
```

Comment:
since we are already adding some type hints to this file, might as well add them to all parameters here. and maybe even to the return value? I am not sure what the return value even is though :)